pytests/stubs/comparisons.pyi

```diff
@@ -0,0 +1,37 @@
+class Eq:
+    def __eq__(self, /, other: Eq) -> bool: ...
+    def __ne__(self, /, other: Eq) -> bool: ...
+    def __new__(cls, /, value: int) -> None: ...
+
+class EqDefaultNe:
+    def __eq__(self, /, other: EqDefaultNe) -> bool: ...
```

Comment:
The `__ne__` method gets "defaulted" on this type, is it just `object.__ne__` (i.e. due to inheritance)?

Or is there actually still a unique `EqDefaultNe.__ne__` method? We might want to special case here if so.

Reply:
This is a bit tricky: PyO3 generates its own `__ne__` default implementation that negates `__eq__`. This is exactly the same behavior as `object.__ne__`. Adding an explicit `EqDefaultNe.__ne__` would be indeed nicer but Mypy already property type check `!=` with the current generated stubs because of the `object.__ne__` fake inheritence. Happy to fix it in a follow up alongside the possible __concat__ vs __add__ conflict

Reply:
Totally makes sense, we can refine in a follow-up 👍